py/gc.c

```diff
@@ -258,10 +258,6 @@ STATIC void gc_sweep(void) {
             case AT_HEAD:
 #if MICROPY_ENABLE_FINALISER
                 if (FTB_GET(block)) {
-                    #if MICROPY_PY_THREAD
-                    // TODO need to think about reentrancy with finaliser code
-                    assert(!"finaliser with threading not implemented");
-                    #endif
```

Comment:
This needs some thought.  It has the same isses that wipy 1.0 originally had with interrupts and threading: when the GC runs it has the GIL, and so if a finaliser runs it also has the GIL.  Then in the VM the finaliser may release the GIL which is not good because the GC is in the middle of running.